metadata-ingestion/src/datahub/ingestion/source/sql/mssql/models.py

```diff
@@ -0,0 +1,706 @@
+import logging
+import re
+import urllib.parse
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
+
+import pydantic
+import sqlalchemy.dialects.mssql
+# This import verifies that the dependencies are available.
+import sqlalchemy_pytds  # noqa: F401
+from pydantic.fields import Field
+from sqlalchemy import create_engine, inspect
+from sqlalchemy.engine.base import Connection
+from sqlalchemy.engine.reflection import Inspector
+
+from datahub.configuration.common import AllowDenyPattern
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.decorators import (
+    SourceCapability,
+    SupportStatus,
+    capability,
+    config_class,
+    platform_name,
+    support_status,
+)
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.sql.sql_common import (
+    BasicSQLAlchemyConfig,
+    SQLAlchemySource,
+    SqlWorkUnit,
+    make_sqlalchemy_uri,
+    register_custom_type,
+)
+from datahub.metadata.schema_classes import (
+    BooleanTypeClass,
+    ChangeTypeClass,
+    UnionTypeClass,
+)
+from datahub.metadata.schema_classes import BooleanTypeClass, UnionTypeClass
+from sqlalchemy.exc import ProgrammingError, ResourceClosedError
+
+from .domains import (
+    JobStep,
+    MSSQLDataFlow,
+    MSSQLDataJob,
+    MSSQLJob,
+    MSSQLProceduresContainer,
+    ProcedureDependency,
+    ProcedureLineageStream,
+    ProcedureParameter,
+    StoredProcedure,
+)
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+register_custom_type(sqlalchemy.dialects.mssql.BIT, BooleanTypeClass)
+register_custom_type(sqlalchemy.dialects.mssql.SQL_VARIANT, UnionTypeClass)
+
+
+class SQLServerConfig(BasicSQLAlchemyConfig):
+    # defaults
+    host_port: str = Field(default="localhost:1433", description="MSSQL host URL.")
+    scheme: str = Field(default="mssql+pytds", description="", hidden_from_schema=True)
+    include_code: bool = Field(
```

Comment:
Yes. I will update naming.
This parameter is disabled by default due to personal preferences for enabling functionality. It is possible to set it by default to enabled if it is considered that it will be convenient to use it.

Reply:
Got it. There is already a feature request for showing job definition at a more convenient place- https://feature-requests.datahubproject.io/p/datajob-definition-tab-similar-to-view-properties. For now, having this on/off flag is good enough, I believe.

Reply:
How does procedure_definition differ from procedure code ?

can you add some tests with jobs and stored procedures ingestion in SQL Server integration tests - https://github.com/datahub-project/datahub/tree/master/metadata-ingestion/tests/integration/sql_server ? It would be helpful during review as well as to monitor any regressions in the code later.